### PR TITLE
Refactor as_patch->as_artist

### LINKED ISSUE #215: Several plotting problems
Some issues with plotting:
 * All points are shown as circles by default.  They should be configurable instead.
 * The point circles fail to display in the specified color (unclear why)
 * the `origin` keyword does nothing in `as_patch` and `plot` methods

There might be others.

### PR DESCRIPTION
**Pull request by keflavich**
Convert `as_patch` to `as_artist` and add a generic `plot` method for pixel regions.

Fixes #215

### CODE REVIEW

**Comment by cdeil** [comment]
In Gammapy, we use regions and call as_patch once:
https://github.com/gammapy/gammapy/search?q=as_patch&unscoped_q=as_patch
That's OK, we can adapt, but we shouldn't change back and forth and cause churn.

So @astrofrog @larrybradley @sushobhana  - do you agree `as_artist` is the better method name than `as_patch` here?

**Comment by keflavich** [comment]
Agreed, we should not go back on this.

We *could* have both `as_patch` and `as_artist` to allow access to patches when patches are the appropriate type.  I don't know if it would be useful, but it would be relatively harmless and would prevent backward compatibility breaking

**Comment by cdeil** [comment]
Suggest to choose either `as_patch` or `as_artist`. There's very few users now, but there will be in the next 10+ years for `astropy.regions`.

So `as_artist` is accurate? All objects returned are MPL Artist objects?

**Comment by keflavich** [comment]
Yes, should be.  Would be good to test this....

**Comment by keflavich** [comment]
Yes, will complete that one first.

File: regions/core/core.py [hunk 1]
```diff
@@ -256,6 +256,26 @@ def as_patch(self, origin=(0, 0), **kwargs):
         """
         raise NotImplementedError
 
+    def plot(self, ax=None, origin=(0, 0), **kwargs):
```
Comment (sushobhana) [inline regions/core/core.py]: This method is already present.

File: regions/shapes/text.py [hunk 2]
```diff
@@ -58,7 +58,7 @@ def to_sky(self, wcs):
         center = pixel_to_skycoord(self.center.x, self.center.y, wcs=wcs)
         return TextSkyRegion(center, self.text)
 
-    def as_patch(self, **kwargs):
+    def as_artist(self, **kwargs):
```
Comment (sushobhana) [inline regions/shapes/text.py]: we should implement this, then the overidden `plot` method can be removed.

File: regions/shapes/point.py [hunk 3]
```diff
@@ -105,15 +105,22 @@ def as_patch(self, origin=(0, 0), **kwargs):
...
         mpl_params.update(kwargs)
+        point = Line2D([self.center.x + origin[0]],
+                       [self.center.y + origin[1]],
+                       **mpl_params)
 
-        return Circle((self.center.x, self.center.y), radius=2, **mpl_params)
+        return point
 
-    def plot(self, origin=(0, 0), ax=None, **kwargs):
+    def plot(self, ax=None, origin=(0, 0), **kwargs):
```
Comment (sushobhana) [inline regions/shapes/point.py]: This should be removed.

File: regions/core/core.py (hunk 1 above)
Comment (keflavich) [reply regions/core/core.py]: good catch

File: regions/shapes/point.py (hunk 3 above)
Comment (keflavich) [reply regions/shapes/point.py]: indeed, it's now redundant with the core plot method

**Comment by keflavich** [comment]
I think we should change the `plot` method to take `axis` as the first argument instead of `origin` being the first argument.  I also think it should be `axis`, not `ax`.  Any objections to me making this change?  Making it here?

**Comment by keflavich** [comment]
I still suggest we change the plot arguments, but let's do that in another PR.

**Pull request merged**